pyqtgraph/colormap.py

```diff
@@ -195,9 +198,84 @@ def getFromColorcet(name):
     cm = ColorMap(
         pos=np.linspace(0.0, 1.0, len(color_list)),
         color=color_list) #, names=color_names)
-    _mapCache[name] = cm
+    if cm is not None:
+        cm.name = name
+        _mapCache[name] = cm
     return cm
+    
+def makeMonochrome(color='green'):
+    """
+    Returns a ColorMap object with a dark to bright ramp and adjustable tint.
+    
+    In addition to neutral, warm or cold grays, imitations of monochrome computer monitors are also
+    available. The following predefined color ramps are available:
+    `neutral`, `warm`, `cool`, `green`, `amber`, `blue`, `red`, `pink`, `lavender`.
+    
+    The ramp can also be specified by a tuple of float values in the range of 0 to 1.
+    In this case `(h, s, l0, l1)` describe hue, saturation, minimum lightness and maximum lightness
+    within the HSL color space. The values `l0` and `l1` can be omitted. They default to 
+    `l0=0.0` and `l1=1.0` in this case.
 
+    Parameters
+    ----------
+    color: str or tuple of floats
+        Color description. Can be one of the predefined identifiers, or a tuple
+        `(h, s, l0, l1)`, `(h, s)` or (`h`).
+        'green', 'amber', 'blue', 'red', 'lavender', 'pink'
+        or a tuple of relative ``(R,G,B)`` contributions in range 0.0 to 1.0
+    """
+    name=f'Monochrome {color}'
+    defaults = {
+        'neutral': (0.00, 0.00, 0.00, 1.00),
+        'warm'   : (0.10, 0.08, 0.00, 0.95),
+        'cool'   : (0.60, 0.08, 0.00, 0.95),
+        'green'  : (0.35, 0.55, 0.02, 0.90),
+        'amber'  : (0.09, 0.80, 0.02, 0.80),
+        'blue'   : (0.58, 0.85, 0.02, 0.95),
+        'red'    : (0.01, 0.60, 0.02, 0.90),
+        'pink'   : (0.93, 0.65, 0.02, 0.95),
+        'lavender': (0.75, 0.50, 0.02, 0.90)
+    }
+    if isinstance(color, str):
+        if color in defaults:
+            h_val, s_val, l_min, l_max = defaults[color]
+        else:
+            valid = ','.join(defaults.keys())
+            raise ValueError(f"Undefined color descriptor '{color}', known values are:\n{valid}")
+    else:
+        s_val = 0.70 # set up default values
+        l_min = 0.00
+        l_max = 1.00
+        if not hasattr(color,'__len__'):
+            h_val = float(color)
+        elif len(color) == 1:
+            h_val = color[0]
+        elif len(color) == 2:
+            h_val, s_val = color
+        elif len(color) == 4:
+            h_val, s_val, l_min, l_max = color
+        else:
+            raise ValueError(f"Invalid color descriptor '{color}'")
+    l_vals = np.linspace(l_min, l_max, num=10)
+    color_list = []
+    for l_val in l_vals:
+        qcol = QtGui.QColor()
+        qcol.setHslF( h_val, s_val, l_val )
+        color_list.append( qcol )
+    return ColorMap( None, color_list, name=name, linearize=True )
+
+def testBarData(length=768, width=32):
```

Comment:
I had to double check, I thought that pytest might mistake this function as a "test" function, but the naming convention doesn't _quite_ match; function names need to be `test_*` (having an underscore).  We can leave this as-is, but if you can think of another function name that might be better; I would encourage a replacement.

Reply:
--> renamed to `colormap.modulatedBarData()`